zipkin/src/main/java/zipkin/Span.java

```diff
@@ -253,11 +253,11 @@ public Builder merge(Span that) {
         } else if (that.duration != null) {
           this.duration = Math.max(this.duration, that.duration);
         }
-      } else { // duration might need to be recalculated, since we have 2 different timestamps
-        long thisEndTs = this.duration != null ? this.timestamp + this.duration : this.timestamp;
-        long thatEndTs = that.duration != null ? that.timestamp + that.duration : that.timestamp;
-        this.timestamp = Math.min(this.timestamp, that.timestamp);
-        this.duration = Math.max(thisEndTs, thatEndTs) - this.timestamp;
+      } else {
+        // We have 2 different timestamps and we don't know which one is authoritative.
```

Comment:
so a suggested fix (which considers client span) would be to move this block below where we iterate through annotations. When we iterate through annotations decide which side is the client span and prefer that somehow..